azurerm/resource_arm_netapp_pool_test.go

```diff
@@ -0,0 +1,235 @@
+)
+
+func TestAccAzureRMNetAppPool_basic(t *testing.T) {
+	resourceName := "azurerm_netapp_pool.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetAppPoolDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetAppPool_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppPoolExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNetAppPool_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	resourceName := "azurerm_netapp_pool.test"
+	ri := tf.AccRandTimeInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetAppPoolDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetAppPool_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppPoolExists(resourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMNetAppPool_requiresImport(ri, testLocation()),
+				ExpectError: testRequiresImportError("azurerm_netapp_pool"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNetAppPool_complete(t *testing.T) {
+	resourceName := "azurerm_netapp_pool.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetAppPoolDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetAppPool_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppPoolExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "service_level", "Standard"),
+					resource.TestCheckResourceAttr(resourceName, "size", "4398046511104"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNetAppPool_update(t *testing.T) {
+	resourceName := "azurerm_netapp_pool.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetAppPoolDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetAppPool_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppPoolExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "service_level", "Premium"),
+				),
+			},
```

Comment:
Can we add another test import step here?

Reply:
Fixed